lang/python/python3-host.mk

```diff
@@ -77,7 +77,7 @@ HOST_PYTHON3_VARS = \
 	LDSHARED="$(HOSTCC) -shared" \
 	CFLAGS="$(HOST_CFLAGS)" \
 	CPPFLAGS="$(HOST_CPPFLAGS) -I$(HOST_PYTHON3_INC_DIR)" \
-	LDFLAGS="$(HOST_LDFLAGS) -lpython$(PYTHON3_VERSION) -Wl$(comma)-rpath$(comma)$(STAGING_DIR_HOSTPKG)/lib" \
+	LDFLAGS="$(HOST_LDFLAGS) -lpython$(PYTHON3_VERSION)" \
```

Comment:
so, at least this one dates back to this commit 11d1b213c28f382bf3d737a05c5375b5f590c2b6   `python: add rpath for host Python extension modules`

then i just just copied this to Python3 at some point in time (to make the diff smaller between Python2 & Python3);

I vaguely remember that the rpath stuff was important to Mac users (when building OpenWrt on MacOS was still a thing);

No idea about it these days

Reply:
Right. Unfortunately I have no macOS system.

The rpath stuff should only be relevant to shared libraries. I've switched most to static. All of python's dependencies are static:

```
> ldd staging_dir/hostpkg/bin/python3
	linux-vdso.so.1 (0x00007fedb1d0f000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fedb1c02000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fedb1a15000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fedb1d11000)

```

Reply:
The reasoning for why setting rpath here was necessary is in https://github.com/openwrt/packages/pull/2489#issuecomment-196417778. Essentially, this is used not for Python but for building Python packages like python-cryptography, to be able to find libffi installed in `$(STAGING_DIR_HOST)/lib`.

I also noted in https://github.com/openwrt/packages/pull/2489#issuecomment-195775981 that this was a fix for buildbot errors and that I wasn't able to reproduce the issue locally, most likely because libffi was installed in the host system.

So to verify that this is no longer necessary, you would need to build python-cryptography (or I believe any other package that depends on python-cffi/host) on a system without libffi installed.

Reply:
that was fixed with https://github.com/openwrt/packages/commit/aba4a294ba35341c2253ff5c0f151613215878c2

there's no libffi.so files installed.

Reply:
Can you try building python-cffi/host, then checking `ldd _cffi_module.so` (I believe _cffi_module.so is installed to `$(STAGING_DIR_HOSTPKG)/lib/python3.11/site-packages`) to see if it still links to libffi.so?

Reply:
```
> ldd staging_dir/hostpkg/lib/python3.11/site-packages/_cffi_backend.cpython-311-x86_64-linux-gnu.so
	linux-vdso.so.1 (0x00007f0c188c9000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f0c1867a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f0c188cb000)

> ls -Rl staging_dir/hostpkg/lib/python3.11/site-packages/ | grep cffi
drwxr-xr-x. 1 mangix mangix    590 Jul  1 16:36 cffi
drwxr-xr-x. 1 mangix mangix    110 Jul  1 16:36 cffi-1.16.0.dist-info
-rwxr-xr-x. 1 mangix mangix 555112 Jul  1 16:36 _cffi_backend.cpython-311-x86_64-linux-gnu.so
staging_dir/hostpkg/lib/python3.11/site-packages/cffi:
-rw-r--r--. 1 mangix mangix  3908 Jul  1 16:36 _cffi_errors.h
-rw-r--r--. 1 mangix mangix 14800 Jul  1 16:36 _cffi_include.h
-rw-r--r--. 1 mangix mangix  5724 Jul  1 16:36 cffi_opcode.py
staging_dir/hostpkg/lib/python3.11/site-packages/cffi/__pycache__:
-rw-r--r--. 1 mangix mangix  6998 Jul  1 16:36 cffi_opcode.cpython-311.opt-1.pyc
-rw-r--r--. 1 mangix mangix  7089 Jul  1 16:36 cffi_opcode.cpython-311.pyc
staging_dir/hostpkg/lib/python3.11/site-packages/cffi-1.16.0.dist-info:

```